pkg/gencosmosdb/cosmosdb/template.go

```diff
@@ -23,12 +23,14 @@ type TemplateClient interface {
 	Delete(context.Context, string, *pkg.Template, *Options) error
 	Query(string, *Query) TemplateIterator
 	QueryAll(context.Context, string, *Query) (*pkg.Templates, error)
+	QueryWithOptions(string, *Query, *Options) TemplateIterator
```

Comment:
I want to keep them separate. As you can go wrong, very fast, very painfully. So it should be scoped. And it is cleaner that having `Query(string, query, nil)` arround the place or `Query(string, Query, Options...)` and dealing with this inside is not nice too.

Reply:
I don't have a problem with API breakage by adding *Options to Query() - it mirrors the fact that Options is in Create, Replace, Delete, etc. already.

Reply:
Still, signature likes this looks much cleaner? Dont you think?

Reply:
Not really ;-)
If we were desperate we could go to `...*Options` but I'm not that bothered

Reply:
...Options looks nasty on the inside
I prefer to keep this way, as it makes it simple to consume. 
If we could merge this I will rebase all other PR's